### PR TITLE
[local_auth]: Renamed `local_auth_ios` to `local_auth_darwin`

### PR DESCRIPTION
**Pull request by OutdatedGuy**
Renames `local_auth_ios` to `local_auth_darwin`

resolves https://github.com/flutter/flutter/issues/141020

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added newxtests to check the change I am making, or this PR is [test-exempt].
- [x] All existin  and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/local_auth/local_auth_darwin/.gitignore
Comment (stuartmorgan) [inline packages/local_auth/local_auth_darwin/.gitignore:1]: Why are there so many new files in a move?

File: packages/local_auth/local_auth_darwin/analysis_options.yaml
Comment (stuartmorgan) [inline packages/local_auth/local_auth_darwin/analysis_options.yaml:1]: Did you run `flutter create` on this package? That should not have been necessary.

This file cannot exist in our repo, and is why a lot of the CI is failing.

File: packages/local_auth/local_auth_darwin/example/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-1024x1024@1x.png
Comment (stuartmorgan) [inline packages/local_auth/local_auth_darwin/example/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-1024x1024@1x.png:1]: Why are all of the app icons new files instead of moves?

File: packages/local_auth/local_auth_darwin/example/ios/Runner.xcodeproj/project.pbxproj
Comment (stuartmorgan) [inline packages/local_auth/local_auth_darwin/example/ios/Runner.xcodeproj/project.pbxproj:1]: Why are there so many changes in this file?

File: packages/local_auth/local_auth_darwin/example/ios/Runner.xcodeproj/project.pbxproj
Comment (stuartmorgan) [inline packages/local_auth/local_auth_darwin/example/ios/Runner.xcodeproj/project.pbxproj:11]: You've broken compilation of the unit tests; see `Mac_arm64 ios_platform_tests_shard_1 master`

File: packages/local_auth/local_auth_darwin/pubspec.yaml
Comment (stuartmorgan) [inline packages/local_auth/local_auth_darwin/pubspec.yaml:2]: Why are there so many style changes in this file?

File: packages/local_auth/local_auth_darwin/pubspec.yaml
Comment (stuartmorgan) [inline packages/local_auth/local_auth_darwin/pubspec.yaml:31]: This violates repository policy and will not pass CI.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]

File: packages/local_auth/local_auth_darwin/pubspec.yaml
Comment (stuartmorgan) [inline packages/local_auth/local_auth_darwin/pubspec.yaml:5]: https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8759389953683448721/+/u/Run_package_tests/CHANGELOG_and_version_validation/stdout

**Review by stuartmorgan — CHANGES_REQUESTED** [review]
I'll do a full re-review sometime this week, but one initial note from CI:

File: packages/local_auth/local_auth_darwin/darwin/local_auth_darwin.podspec
Comment (stuartmorgan) [inline packages/local_auth/local_auth_darwin/darwin/local_auth_darwin.podspec:14]: This line should use the standard boilerplate; we don't list individual authors in this file.

File: packages/local_auth/local_auth_darwin/example/ios/RunnerTests/FLTLocalAuthPluginTests.m
Comment (stuartmorgan) [inline packages/local_auth/local_auth_darwin/example/ios/RunnerTests/FLTLocalAuthPluginTests.m:1]: This file should move to `darwin/Tests/`, and the reference in the Xcode project updated accordingly. (You'll want to change the file's reference type from group-local to project-local in the Xcode UI).

File: packages/local_auth/local_auth_darwin/example/pubspec.yaml
Comment (stuartmorgan) [inline packages/local_auth/local_auth_darwin/example/pubspec.yaml:14]: This should be `local_auth_darwin:` (it was wrong before too, we just didn't catch it).

**Review by stuartmorgan — CHANGES_REQUESTED** [review]
Mostly looks good now! Just some minor things.

File: packages/local_auth/local_auth_darwin/example/ios/RunnerTests/FLTLocalAuthPluginTests.m
Comment (OutdatedGuy) [reply packages/local_auth/local_auth_darwin/example/ios/RunnerTests/FLTLocalAuthPluginTests.m:1]: Hi, I don't use XCode much so I have no idea what you mean. Can someone else do this instead?

Comment (OutdatedGuy) [reply packages/local_auth/local_auth_darwin/example/ios/RunnerTests/FLTLocalAuthPluginTests.m:1]: @jmagman @stuartmorgan @louisehsu Any updates on this?

Comment (stuartmorgan) [reply packages/local_auth/local_auth_darwin/example/ios/RunnerTests/FLTLocalAuthPluginTests.m:1]: @OutdatedGuy We all have a lot of other things we're working on; if you want someone else to do this, you'll need to wait until one of us has time.

Comment (stuartmorgan) [reply packages/local_auth/local_auth_darwin/example/ios/RunnerTests/FLTLocalAuthPluginTests.m:1]: I've pushed this change.

**Comment by stuartmorgan** [comment]
This will need to be reconciled with main due to the change from iOS 11 to iOS 12 as a minimum version.

**Review by jmagman — APPROVED** [review]
LGTM!  Can you rebase and fix up the merge conflicts again, then we're ready to merge.  Really appreciate you tackling this, looking forward to the macOS support on top of this!

File: packages/local_auth/local_auth_darwin/darwin/local_auth_darwin.podspec
Comment (jmagman) [inline packages/local_auth/local_auth_darwin/darwin/local_auth_darwin.podspec:2]: Actually this needs the `PrivacyInfo.xcprivacy` pulled in and `resource_bundles` path updated:
```
 -> local_auth_darwin (0.0.1)
    - NOTE  | url: The URL (https://pub.dev/packages/local_auth_darwin) is not reachable.
    - ERROR | [iOS] file patterns: The `resource_bundles` pattern for `local_auth_ios_privacy` did not match any file.
```

See https://github.com/flutter/packages/pull/5846/files#diff-3224e4fc5e9a47bae7d7c5dcc243e869174b97b0b4e8572a6e58fde8c0b3f5d5

**Comment by jmagman** [comment]
Can you also switch
```
 Unknown directory "packages/local_auth/local_auth_ios" in CODEOWNERS
```
https://github.com/flutter/packages/blob/316bde0b3908ee4b6f6928476ec1b4ecba393e39/CODEOWNERS#L91

Hopefully almost done 🤞

**Comment by jmagman** [comment]
@stuartmorgan can you take a look at this again?  Blocked on your change request.

**Review by stuartmorgan — APPROVED** [review]
LGTM!

**Pull request merged**
